ldap/src/main/java/org/springframework/security/ldap/search/FilterBasedLdapUserSearch.java

```diff
@@ -118,9 +117,6 @@ public DirContextOperations searchForUser(String username) {
 					+ this);
 		}
 
-		SpringSecurityLdapTemplate template = new SpringSecurityLdapTemplate(
-				contextSource);
-
 		template.setSearchControls(searchControls);
```

Comment:
We should not be setting a value on a member variable because it introduces a race condition. We either need to to keep this at a method level or we need to restructure the code so that updates to searchControls via the setter methods update the template.

Reply:
@rwinch Thanks for the feedback! I overlooked that part and thought it's immutable somehow. Sorry for the noise. I'm closing this for now.

Reply:
@rwinch BTW based on the your comment, it looks there's visibility problem with `searchControls` in the current implementation. Am I missing something?

Reply:
I think you are right. It should have been structured differently, but it is likely due to historical reasons.

Reply:
@rwinch I see. I'd like to try but I don't have much knowledge on the class to restructure it 😅